addons/payment/controllers/portal.py

```diff
@@ -140,6 +141,7 @@ def payment_pay(
             'landing_route': '/payment/confirmation',
             'partner_is_different': partner_is_different,
             'invoice_id': invoice_id,
+            'res_company': res_company,
```

Comment:
Please swap these two. It helps my team immediately identify when which parameters are used ;)
Also, a comment never hurts for corner cases like this one.
```suggestion
            'res_company': company,  # Display the correct logo in a multi-company environment.
            'partner_is_different': partner_is_different,
            'invoice_id': invoice_id,
```